web-gateway/src/main/java/controllers/ItemController.java

```diff
@@ -166,65 +167,86 @@ private ItemData fromForm(ItemForm itemForm) {
             }
         });
     }
-
     public CompletionStage<Result> getItem(String itemId) {
         return doGetItem(ctx(), itemId, formFactory.form(BidForm.class));
     }
 
     private CompletionStage<Result> doGetItem(Http.Context ctx, String itemId, Form<BidForm> bidForm) {
         return requireUser(ctx, user -> loadNav(user).thenComposeAsync(nav -> {
+
             UUID itemUuid = UUID.fromString(itemId);
             CompletionStage<Item> itemFuture = itemService.getItem(itemUuid)
                     .handleRequestHeader(authenticate(user)).invoke();
             CompletionStage<PSequence<Bid>> bidHistoryFuture = bidService.getBids(itemUuid)
                     .handleRequestHeader(authenticate(user)).invoke();
-            return itemFuture.thenCombineAsync(bidHistoryFuture, (item, bidHistory) -> {
 
-                if (item.getStatus() == ItemStatus.CREATED && !item.getCreator().equals(user)) {
-                    return forbidden();
-                }
+                return itemFuture.thenCombineAsync(bidHistoryFuture, (item, bidHistory) -> {
+
+                    User seller = null;
+                    Optional<User> winner = Optional.empty();
+
+                    try {
+                        seller = userService.getUser(item.getCreator()).invoke().toCompletableFuture().get();
```

Comment:
Since `AbstractController` requires the `UserService` for _every_ request (to lookup the current user and the list of users) there would not be much point adding a fallback here until that is addressed somehow. I don't think that dependencies are something that can or should be eliminated entirely. We also have to be concerned about the complexity of the implementation.